### PR TITLE
[misc] nginx/compose: use docker-compose spec v3

### PR DESCRIPTION
**Pull request by das7pad**
### Proposed changes

Migrate the docker-compose file of the nginx auth example from v1 to v3.

### Related issues (optional)

Followup for #9270 

---

NOTE1: bcrypt is still not supported by the `nginx:latest` docker image

NOTE2: I tested the example using this minimal nginx config to omit the ssl setup:

```
events {
  worker_connections  1024;
}

http {
  upstream docker-registry {
    server registry:5000;
  }
  server {
    server_name _;
    listen 443;

    auth_basic "Registry realm";
    auth_basic_user_file /etc/nginx/conf.d/nginx.htpasswd;

    location /v2/ {
      proxy_pass http://docker-registry;
    }
  }
}
```

```
# curl testuser:testpassword@127.0.0.1:5043/v2/ -i
HTTP/1.1 200 OK
Server: nginx/1.17.3
Date: Thu, 12 Sep 2019 23:30:14 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 2
Connection: keep-alive
Docker-Distribution-Api-Version: registry/2.0
X-Content-Type-Options: nosniff
```

### CODE REVIEW

File: registry/recipes/nginx.md [hunk 1]
```diff
@@ -166,23 +166,24 @@ Review the [requirements](/registry/recipes/index.md#requirements), then follow
...
-        - registry:registry
-      volumes:
-        - ./auth:/etc/nginx/conf.d
-        - ./auth/nginx.conf:/etc/nginx/nginx.conf:ro
-
-    registry:
-      image: registry:2
-      volumes:
-        - ./data:/var/lib/registry
+    version: "3"
```
Comment (usha-mandya) [inline registry/recipes/nginx.md:169]: @glours How do we represent the Compose spec here instead of the legacy version 3?

**Comment by das7pad** [comment]
In regards to https://github.com/docker/docs/pull/9441#discussion_r942268827 this is likely still of interest.

/remove-lifecycle stale

**Comment by das7pad** [comment]
In regards to https://github.com/docker/docs/pull/9441#discussion_r942268827 this is likely still of interest.

/remove-lifecycle stale

File: registry/recipes/nginx.md (hunk 1 above)
Comment (das7pad) [reply registry/recipes/nginx.md:173]: Is there any update here?

Comment (thaJeztah) [reply registry/recipes/nginx.md:173]: I believe we still need to remove these docs from the repository (as the docs were transferred to the upstream project); if someone can open a pull request for the upstream instead?

Upstream file is in this repo and location on GitHub; https://github.com/distribution/distribution/blob/main/docs/recipes/nginx.md

Comment (das7pad) [reply registry/recipes/nginx.md:173]: Thanks for the hint! I've opened https://github.com/distribution/distribution/pull/3872. I'll close this one then.

**Pull request closed without merging**